test/src/test/java/hudson/slaves/JNLPLauncherRealTest.java

```diff
@@ -104,7 +104,7 @@ public void run(JenkinsRule r) throws Throwable {
             p.setAssignedNode(agent);
             FreeStyleBuild b = r.buildAndAssertSuccess(p);
             if (webSocket) {
-                assertThat(agent.toComputer().getSystemProperties().get("java.class.path"), is(new File(r.jenkins.root, "agent.jar").getAbsolutePath()));
+                assertThat(agent.toComputer().getSystemProperties().get("java.class.path").toString(), endsWith("agent.jar"));
```

Comment:
Assertion introduced in #6780, if I recall correctly just to somehow use the `Channel`, and made more brittle in #7192. Does not need to check the exact path, or even really this property, it is just exercising Remoting.